prov/mrail/src/mrail.h

```diff
@@ -172,13 +172,21 @@ struct mrail_av {
 	ofi_atomic32_t index;
 };
 
-struct mrail_peer_addr {
+struct mrail_peer_info {
+	struct slist	ooo_recv_queue;
 	fi_addr_t	addr;
 	uint32_t	seq_no;
+	uint32_t	expected_seq_no;
```

Comment:
This patch should have actually been at least 3.  1. rename mrail_peer_addr to mrail_peer_info.  2. replace mrail ep lock with util ep lock.  3. handle out of order messages.

Maybe I wasn't clear in my previous message.  What a series should not contain is a patch that updates or fixes an earlier patch in the series.  Don't respin this patch to break it back apart.  Just update based on the other comments.

Reply:
Understood. I'll update this with your comments.